cmd/kwild/server/build.go

```diff
@@ -428,10 +428,10 @@ func buildDB(d *coreDependencies, closer *closeFuncs) *pg.DB {
 // DB restoration from snapshot is skipped in the following scenarios:
 //   - If the DB is already initialized (i.e this is not a new node)
 //   - If the genesis apphash is not specified
-//   - If statesync is enabled. Statesync will take care of rapildly syncing the database
+//   - If statesync is enabled. Statesync will take care of syncing the database
 //     to the network state using statesync snapshots.
 func restoreDB(d *coreDependencies) {
-	if isDbInitialized(d) || d.genesisCfg.DataAppHash == nil || d.cfg.ChainCfg.StateSync.Enable {
+	if d.cfg.ChainCfg.StateSync.Enable || len(d.genesisCfg.DataAppHash) == 0 || isDbInitialized(d) {
```

Comment:
@charithabandi This is the short-circuit behavior I mentioned in a recent review.  The boolean operations  are evaluated left-to-right, but not necessarily all of them.  With OR, it stops when it hits the first `true`.  With AND, it stops when it hits the first `false`.  Sorry if you were already aware; just wanted to be explicit.

Reply:
lol, Can't believe I wrote it this way